packages/jaeger-ui/src/components/TracePage/TraceStatistics/tableValues.tsx

```diff
@@ -13,88 +13,40 @@
+  // To work around that, we multiply start/end times by 10 and subtract one from the end.
+  // So instead of [1-10] we get [10-99]. This makes the intervals work like half-open.
+  const spanRange = new DRange(10 * span.startTime, 10 * (span.startTime + span.duration) - 1);
+  // Only keep CHILD_OF spans. FOLLOWS_FROM spans do not block the parent.
+  const children = allSpans.filter(
+    each =>
+      span.childSpanIds.includes(each.spanID) &&
+      each.references[0].spanID === span.spanID &&
+      each.references[0].refType === 'CHILD_OF'
+  );
+  children.forEach(child => {
+    spanRange.subtract(10 * child.startTime, 10 * (child.startTime + child.duration) - 1);
   });
-  if (temp !== undefined) {
-    result = { duration: temp.duration, lowestStartTime: temp.relativeStartTime };
-  } else {
-    result = { duration: -1, lowestStartTime: -1 };
-  }
-  return result;
-}
-
-/**
- * Determines whether the cut spans belong together and then calculates the duration.
- */
-function getDuration(lowestStartTime: number, duration: number, allOverlay: Span[]) {
-  let durationChange = duration;
-  let didDelete = false;
-  for (let i = 0; i < allOverlay.length; i++) {
-    if (lowestStartTime + durationChange >= allOverlay[i].relativeStartTime) {
-      if (lowestStartTime + durationChange < allOverlay[i].relativeStartTime + allOverlay[i].duration) {
-        const tempDuration =
-          allOverlay[i].relativeStartTime + allOverlay[i].duration - lowestStartTime + durationChange;
-        durationChange = tempDuration;
-      }
-      allOverlay.splice(i, 1);
-      didDelete = true;
-      break;
-    }
-  }
-  const result = { allOverlay, duration: durationChange, didDelete };
-  return result;
-}
-
-/**
- * Return the selfTime of overlay spans.
- */
-function onlyOverlay(allOverlay: Span[], allChildren: Span[], tempSelf: number, span: Span) {
-  let tempSelfChange = tempSelf;
-  let duration = 0;
-  let resultGetDuration = { allOverlay, duration, didDelete: false };
-  const noOverlay = _.difference(allChildren, allOverlay);
-  let lowestStartTime = 0;
-  let totalDuration = 0;
-  const result = getLowestStartTime(allOverlay);
-  lowestStartTime = result.lowestStartTime;
-  duration = result.duration;
-
-  do {
-    resultGetDuration = getDuration(lowestStartTime, duration, resultGetDuration.allOverlay);
-    if (!resultGetDuration.didDelete && resultGetDuration.allOverlay.length > 0) {
-      totalDuration = resultGetDuration.duration;
-      const temp = getLowestStartTime(resultGetDuration.allOverlay);
-      lowestStartTime = temp.lowestStartTime;
-      duration = temp.duration;
-    }
-  } while (resultGetDuration.allOverlay.length > 1);
-  duration = resultGetDuration.duration + totalDuration;
-  // no cut is observed
-  for (let i = 0; i < noOverlay.length; i++) {
-    duration += noOverlay[i].duration;
-  }
-  tempSelfChange += span.duration - duration;
-
-  return tempSelfChange;
+  return Math.round(spanRange.length / 10);
```

Comment:
btw, I don't think rounding is necessary, length should always be divisible by 10.